### PR TITLE
feat: implements convert mnemonic to privateKey

### PR DESCRIPTION
**Pull request by gyuguen**
Added a function to convert to privateKey using `mnemonic` and `hdPath`

### CODE REVIEW

**Comment by gyuguen** [comment]
My knowledge was short and I couldn't run the test again.
I'll check it out here.

**Review by H4NLee — APPROVED** [review]
It seems you applied different code style compared to the existing code.
(e.g. space within braces, indent)
I think it would be good to use consistent code style :)

File: docs/examples.md [hunk 1]
```diff
@@ -144,3 +144,23 @@ assertIsDeliverTxSuccess(result);
...
+const body = "testBody";
+const hashedBody = sha256(new TextEncoder().encode(body));
+const mnemonic = "bulb rail ...";
+const hdPath = stringToPath("m/44'/371'/0'/0/0");
+
+const privateKey = await Secp256k1.parseMnemonicToPrivateKey(mnemonic, hdPath);
+const {pubkey} = await CryptoSecp256k1.makeKeypair(privateKey);
+
+const signature = await CryptoSecp256k1.createSignature(hashedBody, privateKey)
+console.log(CryptoSecp256k1.verifySignature(signature, hashedBody, pubkey))
```
Comment (youngjoon-lee) [inline docs/examples.md:165]:
```suggestion
const mnemonic = "bulb rail ...";
const hdPath = stringToPath("m/44'/371'/0'/0/0");

const privateKey = await Secp256k1.parseMnemonicToPrivateKey(mnemonic, hdPath);
const {pubkey} = await CryptoSecp256k1.makeKeypair(privateKey);
```
I recommend you to just show how to convert a mnemonic to a keypair.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@medibloc/panacea-js",
-  "version": "2.0.1",
+  "version": "2.0.2",
```
Comment (youngjoon-lee) [inline package.json:3]: Please revert this version number for now. After merging this PR, I will create a release commit.

File: src/crypto/secp256k1.ts [hunk 3]
```diff
@@ -1,20 +1,27 @@
...
 
 export class Secp256k1 {
-  static generatePrivateKey(): Uint8Array {
-    let privKey
-    do {
-      privKey = randomBytes(32)
-    } while (!ecc.privateKeyVerify(privKey));
-    return privKey;
-  }
+    static generatePrivateKey(): Uint8Array {
```
Comment (youngjoon-lee) [inline src/crypto/secp256k1.ts:6]: As Hansol said, please use 2 spaces for indentations.

File: src/crypto/secp256k1.ts [hunk 4]
```diff
@@ -1,20 +1,27 @@
 import ecc from "secp256k1";
-import { randomBytes } from "crypto";
+import {randomBytes} from "crypto";
+import {Bip39, EnglishMnemonic, Slip10, Slip10Curve, Slip10RawIndex} from "@cosmjs/crypto"
```
Comment (youngjoon-lee) [inline src/crypto/secp256k1.ts:3]:
```suggestion
import { randomBytes } from "crypto";
import { Bip39, EnglishMnemonic, Slip10, Slip10Curve, Slip10RawIndex } from "@cosmjs/crypto"
```
Also, please insert spaces inside brackets. 

Javascript/Typescripts sucks on this formatting. VSCode and Jetbrains have their own way to configure this. To enhance this, I will find some 'lint' practices.

File: src/crypto/secp256k1.ts [hunk 5]
```diff
@@ -1,20 +1,27 @@
...
-    return ecc.ecdsaSign(data32, privKey).signature;
-  }
+    static sign(data32: Uint8Array, privKey: Uint8Array): Uint8Array {
+        return ecc.ecdsaSign(data32, privKey).signature;
+    }
+
+    static async parseMnemonicToPrivateKey(mnemonic: string, hdPath: readonly Slip10RawIndex[]): Promise<Uint8Array> {
+        const mnemonicChecked = new EnglishMnemonic(mnemonic)
+        const seed = await Bip39.mnemonicToSeed(mnemonicChecked, "")
+        return Slip10.derivePath(Slip10Curve.Secp256k1, seed, hdPath).privkey
```
Comment (youngjoon-lee) [inline src/crypto/secp256k1.ts:25]:
```suggestion
        const mnemonicChecked = new EnglishMnemonic(mnemonic);
        const seed = await Bip39.mnemonicToSeed(mnemonicChecked, "");
        return Slip10.derivePath(Slip10Curve.Secp256k1, seed, hdPath).privkey;
```
Also, please use semicolons.

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,8 @@
 ## Unreleased
 
 TBD
+- [\#64](https://github.com/medibloc/panacea-js/pull/64) feat: implements convert mnemonic to privateKey of Secp256k1
```
Comment (youngjoon-lee) [inline CHANGELOG.md:6]:
```suggestion
- [\#64](https://github.com/medibloc/panacea-js/pull/64) feat: implement a function to convert a mnemonic to a secp256k1 private key
```

**Review by youngjoon-lee — APPROVED** [review]
lgtm

File: docs/examples.md [hunk 7]
```diff
@@ -144,3 +144,18 @@ assertIsDeliverTxSuccess(result);
...
 console.log(didDocumentWithSeq);
 ```
+
+## Convert mnemonic to Secp256k1 privateKey
+
+```ts
+import {Secp256k1 as CryptoSecp256k1, stringToPath, sha256} from "@cosmjs/crypto";
+import {panacead} from "../testutils";
+import {TextEncoder} from "util";
+import {Secp256k1} from "./secp256k1";
```
Comment (youngjoon-lee) [inline docs/examples.md:154]:
```suggestion
import { Secp256k1 as CryptoSecp256k1, stringToPath } from "@cosmjs/crypto";
import { Secp256k1 } from "@medibloc/panacea-js";
```

File: package.json (hunk 2 above)
Comment (gyuguen) [reply package.json]: Okay. I fixed it.

File: src/crypto/secp256k1.ts (hunk 3 above)
Comment (gyuguen) [reply src/crypto/secp256k1.ts]: Yes, I fixed it.
Intellij applied 4 spaces by default.

File: src/crypto/secp256k1.ts (hunk 4 above)
Comment (gyuguen) [reply src/crypto/secp256k1.ts]: I fixed it.
This also changed the formatter in Intellij.

File: src/crypto/secp256k1.ts (hunk 5 above)
Comment (gyuguen) [reply src/crypto/secp256k1.ts]: Thanks, I fixed it.

**Review by youngjoon-lee — APPROVED** [review]

**Comment by youngjoon-lee** [comment]
I'm gonna improve our lint script and also make it run in CI.

**Comment by gyuguen** [comment]
> I'm gonna improve our lint script and also make it run in CI.

Yep. Very good.

**Comment by gyuguen** [comment]
@youngjoon-lee 
Please deploy the version with this feature to npm. (__)

**Comment by youngjoon-lee** [comment]
> @youngjoon-lee Please deploy the version with this feature to npm. (__)

yup. will do it now.

**Pull request merged**